api/models/user.go

```diff
@@ -3,55 +3,50 @@ package models
 import (
 	"encoding/json"
 	"errors"
+	
 	"time"
+	"gopkg.in/go-playground/validator.v9"
 
 	"github.com/jinzhu/gorm"
 	uuid "github.com/satori/go.uuid"
 )
 
-type myTime time.Time
-
-var _ json.Unmarshaler = &myTime{}
-
-func (mt *myTime) UnmarshalJSON(bs []byte) error {
-	var s map[string]string
-	err := json.Unmarshal(bs, &s)
-	if err != nil {
-		return err
-	}
-
-	t, err := time.Parse("2008-19-12", "2008-19-12")
-	if err != nil {
-		return err
-	}
-
-	*mt = myTime(t)
-	return nil
-}
-
 // User represents the user.
 type User struct {
 	ID          int       `gorm:"primary_key"`
 	UUID        uuid.UUID `json:"uuid"`
-	AccessLevel int
-	FirstName   string `json:"first_name" binding:"required"`
-	LastName    string `json:"last_name"`
-	Email       string `json:"email"`
-	Password    string `json:"pass"`
-	DateOfBirth myTime `json:"birth_date"`
+	AccessLevel int		`json:"access_level" Usage:"oneof=0 1"`
+	FirstName   string `json:"first_name" validate:"required,min=2,excludesall=0x20" Usage:"alpha"`
+	LastName    string `json:"last_name" validate:"required,min=2,excludesall=0x20" Usage:"alpha"`
+	Email       string `json:"email" validate:"required,email" Usage:"unique"`
+	Password    string `json:"pass" validate:"required" Usage:"eqfield=confirm_password"`
+	// DateOfBirth time.Time `json:"birth_date" validate:"required,adult"`
```

Comment:
Pourquoi là c'est en commentaire ?

Reply:
It’s the custom method for the date of birth field and it’s generating some errors (//FIXME)